programs/drift/src/state/lp_pool.rs

```diff
@@ -0,0 +1,1618 @@
+            );
+
+            // sum up crypto deltas (notional exposures for all non-stablecoins)
+            if constituent.constituent_index != self.usdc_consituent_index
+                && constituent.constituent_derivative_index != self.usdc_consituent_index as i16
+            {
+                let constituent_target_notional = constituent_target_base
+                    .get(constituent.constituent_index as u32)
+                    .target_base
+                    .cast::<i128>()?
+                    .safe_mul(constituent.last_oracle_price.cast::<i128>()?)?
+                    .safe_div(10_i128.pow(constituent.decimals as u32))?
+                    .cast::<i64>()?;
+                crypto_delta = crypto_delta.safe_add(constituent_target_notional.cast()?)?;
+            }
+            aum = aum.safe_add(constituent_aum)?;
+        }
+
+        msg!("Aum before quote owed from lp pool: {}", aum);
+
+        for cache_datum in amm_cache.iter() {
+            aum -= cache_datum.quote_owed_from_lp_pool as i128;
+        }
+
+        let aum_u128 = aum.max(0i128).cast::<u128>()?;
+        self.last_aum = aum_u128;
+        self.last_aum_slot = slot;
+        self.last_aum_ts = now;
+
+        Ok((aum_u128, crypto_delta, derivative_groups))
+    }
+}
+
+#[zero_copy(unsafe)]
+#[derive(Default, Eq, PartialEq, Debug, BorshDeserialize, BorshSerialize)]
+#[repr(C)]
+pub struct BLPosition {
+    /// The scaled balance of the position. To get the token amount, multiply by the cumulative deposit/borrow
+    /// interest of corresponding market.
+    /// precision: token precision
+    pub scaled_balance: u128,
+    /// The cumulative deposits/borrows a user has made into a market
+    /// precision: token mint precision
+    pub cumulative_deposits: i64,
+    /// The market index of the corresponding spot market
+    pub market_index: u16,
+    /// Whether the position is deposit or borrow
+    pub balance_type: SpotBalanceType,
+    pub padding: [u8; 5],
+}
+
+impl SpotBalance for BLPosition {
+    fn market_index(&self) -> u16 {
+        self.market_index
+    }
+
+    fn balance_type(&self) -> &SpotBalanceType {
+        &self.balance_type
+    }
+
+    fn balance(&self) -> u128 {
+        self.scaled_balance as u128
+    }
+
+    fn increase_balance(&mut self, delta: u128) -> DriftResult {
+        self.scaled_balance = self.scaled_balance.safe_add(delta)?;
+        Ok(())
+    }
+
+    fn decrease_balance(&mut self, delta: u128) -> DriftResult {
+        self.scaled_balance = self.scaled_balance.safe_sub(delta)?;
+        Ok(())
+    }
+
+    fn update_balance_type(&mut self, balance_type: SpotBalanceType) -> DriftResult {
+        self.balance_type = balance_type;
+        Ok(())
+    }
+}
+
+impl BLPosition {
+    pub fn get_token_amount(&self, spot_market: &SpotMarket) -> DriftResult<u128> {
+        get_token_amount(self.scaled_balance, spot_market, &self.balance_type)
+    }
+}
+
+#[account(zero_copy(unsafe))]
```

Comment:
might make it's own file if it's account. Also we could consider adding `lp` dir in state if that makes it easier to navigate